crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1993,37 +1993,45 @@ pub async fn retrieve_payment_method(
 pub async fn delete_payment_method(
     state: &routes::AppState,
     merchant_account: domain::MerchantAccount,
-    pm: api::PaymentMethodId,
+    pm_id: api::PaymentMethodId,
 ) -> errors::RouterResponse<api::PaymentMethodDeleteResponse> {
-    let (_, supplementary_data) =
-        vault::Vault::get_payment_method_data_from_locker(state, &pm.payment_method_id).await?;
-    let payment_method_id = supplementary_data
-        .payment_method_id
-        .map_or(Err(errors::ApiErrorResponse::PaymentMethodNotFound), Ok)?;
-    let pm = state
-        .store
-        .delete_payment_method_by_merchant_id_payment_method_id(
-            &merchant_account.merchant_id,
-            &payment_method_id,
-        )
+    let db = &*state.store;
+    let key = db
+        .find_payment_method(pm_id.payment_method_id.as_str())
         .await
         .to_not_found_response(errors::ApiErrorResponse::PaymentMethodNotFound)?;
 
-    if pm.payment_method == enums::PaymentMethod::Card {
-        let response =
-            delete_card_from_locker(state, &pm.customer_id, &pm.merchant_id, &payment_method_id)
-                .await?;
+    if key.payment_method == enums::PaymentMethod::Card {
+        let response = delete_card_from_locker(
+            state,
+            &key.customer_id,
+            &key.merchant_id,
+            pm_id.payment_method_id.as_str(),
+        )
+        .await?;
+
         if response.status == "SUCCESS" {
-            print!("Card From locker deleted Successfully")
+            db.delete_payment_method_by_merchant_id_payment_method_id(
+                &merchant_account.merchant_id,
+                pm_id.payment_method_id.as_str(),
+            )
+            .await
+            .to_not_found_response(errors::ApiErrorResponse::PaymentMethodNotFound)?;
+
+            Ok(services::ApplicationResponse::Json(
+                api::PaymentMethodDeleteResponse {
+                    payment_method_id: key.payment_method_id,
+                    deleted: true,
+                    status: Some(response.status),
+                },
+            ))
         } else {
-            print!("Error: Deleting Card From Locker")
+            Err(errors::ApiErrorResponse::UnprocessableEntity {
+                entity: response.error_message.unwrap_or_default(),
+            })?
         }
-    };
-
-    Ok(services::ApplicationResponse::Json(
-        api::PaymentMethodDeleteResponse {
-            payment_method_id: pm.payment_method_id,
-            deleted: true,
-        },
-    ))
+    } else {
+        // This part of code is like a place holder and the control will never come here since everything is handled above
+        Err(errors::ApiErrorResponse::PaymentMethodNotFound)?
```

Comment:
In cases where the payment method isn't `Card`. we should have certain handling in place to redact or remove the payment method. 
What do you think @jarnura, @lsampras?

Reply:
No need to handle this in the PR, but would be good to create a issue if necessary